File: software/examples/vbnet/ExampleCallback.vb
Imports System
Imports Tinkerforge

Module ExampleCallback
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Accelerometer Bricklet 2.0

    ' Callback subroutine for acceleration callback
    Sub AccelerationCB(ByVal sender As BrickletAccelerometerV2, ByVal x As Integer, _
                       ByVal y As Integer, ByVal z As Integer)
        Console.WriteLine("Acceleration [X]: " + (x/10000.0).ToString() + " g")
        Console.WriteLine("Acceleration [Y]: " + (y/10000.0).ToString() + " g")
        Console.WriteLine("Acceleration [Z]: " + (z/10000.0).ToString() + " g")
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim a As New BrickletAccelerometerV2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register acceleration callback to subroutine AccelerationCB
        AddHandler a.AccelerationCallback, AddressOf AccelerationCB

        ' Set period for acceleration callback to 1s (1000ms)
        a.SetAccelerationCallbackConfiguration(1000, False)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/delphi/ExampleCallback.pas
program ExampleCallback;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletAccelerometerV2;

type
  TExample = class
  private
    ipcon: TIPConnection;
    a: TBrickletAccelerometerV2;
  public
    procedure AccelerationCB(sender: TBrickletAccelerometerV2; const x: longint;
                             const y: longint; const z: longint);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change XYZ to the UID of your Accelerometer Bricklet 2.0 }

var
  e: TExample;

{ Callback procedure for acceleration callback }
procedure TExample.AccelerationCB(sender: TBrickletAccelerometerV2; const x: longint;
                                  const y: longint; const z: longint);
begin
  WriteLn(Format('Acceleration [X]: %f g', [x/10000.0]));
  WriteLn(Format('Acceleration [Y]: %f g', [y/10000.0]));
  WriteLn(Format('Acceleration [Z]: %f g', [z/10000.0]));
  WriteLn('');
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  a := TBrickletAccelerometerV2.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Register acceleration callback to procedure AccelerationCB }
  a.OnAcceleration := {$ifdef FPC}@{$endif}AccelerationCB;

  { Set period for acceleration callback to 1s (1000ms) }
  a.SetAccelerationCallbackConfiguration(1000, false);

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
